File: src/main/java/fr/cph/crypto/client/TickerClient.java
package fr.cph.crypto.client;

import java.util.List;
import java.util.Optional;

import fr.cph.crypto.domain.Currency;
import fr.cph.crypto.domain.Ticker;

public interface TickerClient {

	List<Ticker> getTickers(Currency currency, String... tickers);

	List<Ticker> getTickers(Currency currency, List<String> tickers);

	Optional<Ticker> getTicker(Currency currency, String ticker);
}

File: src/main/kotlin/fr/cph/crypto/client/TickerClient.kt
package fr.cph.crypto.client

import java.util.Optional

import fr.cph.crypto.domain.Currency
import fr.cph.crypto.domain.Ticker

interface TickerClient {

    fun getTickers(currency: Currency, vararg tickers: String): List<Ticker>

    fun getTickers(currency: Currency, tickers: List<String>): List<Ticker>

    fun getTicker(currency: Currency, ticker: String): Optional<Ticker>
}
